batched/dense/impl/KokkosBatched_SVD_Serial_Impl.hpp

```diff
@@ -41,6 +54,15 @@ template <typename AViewType, typename SViewType, typename WViewType>
 KOKKOS_INLINE_FUNCTION int SerialSVD::invoke(SVD_S_Tag, const AViewType &A,
                                              const SViewType &sigma,
                                              const WViewType &work) {
+  static_assert(Kokkos::is_view_v<AViewType> && AViewType::rank == 2,
+                "SVD: A must be a rank-2 view");
+  static_assert(Kokkos::is_view_v<SViewType> && SViewType::rank == 1,
+                "SVD: s must be a rank-1 view");
+  static_assert(Kokkos::is_view_v<WViewType> && WViewType::rank == 1,
+                "SVD: W must be a rank-1 view");
+  static_assert(
+      !std::is_same_v<typename WViewType::array_layout, Kokkos::LayoutStride>,
```

Comment:
@e10harvey The new test in this PR tests with A, U, S and V being LayoutStride. It's only the work array that must be contiguous, a convention from some of Kyungjoo's older functions like QR, eigendecomposition, etc.

For checking the contiguous-ness of a rank-1 view at compile time, I still prefer to just blacklist LayoutStride just because that's more consistent with the rest of the library.